sonic_db_config/db_config.go

```diff
@@ -1,135 +1,237 @@
+}
+
+func GetDbHostName(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	hostname, ok := inst["hostname"]
+	if !ok {
+		panic(fmt.Errorf("'hostname' is not a valid field in database_config.json file!"))
+	}
+	return hostname.(string)
+}
+
+func GetDbPort(db_name string, ns string) int {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	port, ok := inst["port"]
+	if !ok {
+		panic(fmt.Errorf("'port' is not a valid field in database_config.json file!"))
+	}
+	return int(port.(float64))
+}
+
+func GetDbTcpAddr(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	hostname := GetDbHostName(db_name, ns)
+	port := GetDbPort(db_name, ns)
+	return hostname + ":" + strconv.Itoa(port)
+}
+
+func DbGetNamespaceAndConfigFile(ns_to_cfgfile_map map[string]string) {
+	data, err := io.ReadFile(SONIC_DB_GLOBAL_CONFIG_FILE)
+	if err == nil {
+		//Ref:https://stackoverflow.com/questions/18537257/how-to-get-the-directory-of-the-currently-running-file
+		dir, err := filepath.Abs(filepath.Dir(SONIC_DB_GLOBAL_CONFIG_FILE))
+		if err != nil {
+			panic(err)
+		}
+		var sonic_db_global_config = make(map[string]interface{})
+		err = json.Unmarshal([]byte(data), &sonic_db_global_config)
+		if err != nil {
+			panic(err)
+		}
+		var db_include_file string
+		for _, entry := range sonic_db_global_config["INCLUDES"].([]interface{}) {
+			ns, ok := entry.(map[string]interface{})["namespace"]
+			if !ok {
+				ns = SONIC_DEFAULT_NAMESPACE
+			}
+			_, ok = ns_to_cfgfile_map[ns.(string)]
+			if ok {
+				panic(fmt.Errorf("Global Database config file is not valid(multiple include for same namespace!"))
+			}
+			//Ref:https://www.geeksforgeeks.org/filepath-join-function-in-golang-with-examples/
+			db_include_file = filepath.Join(dir, entry.(map[string]interface{})["include"].(string))
+			ns_to_cfgfile_map[ns.(string)] = db_include_file
+		}
+		if len(ns_to_cfgfile_map) > 1 {
+			sonic_db_multi_namespace = true
+		} else {
+			sonic_db_multi_namespace = false
+		}
+
+	} else if errors.Is(err, os.ErrNotExist) {
+		// Ref: https://stackoverflow.com/questions/23452157/how-do-i-check-for-specific-types-of-error-among-those-returned-by-ioutil-readfi
+		ns_to_cfgfile_map[SONIC_DEFAULT_NAMESPACE] = SONIC_DB_CONFIG_FILE
+		sonic_db_multi_namespace = false
+	} else {
+		panic(err)
+	}
 }
 
 func DbInit() {
-    if sonic_db_init {
-        return
-    }
-    data, err := io.ReadFile(SONIC_DB_CONFIG_FILE)
-    if err != nil {
-        panic(err)
-    } else {
-        err = json.Unmarshal([]byte(data), &sonic_db_config)
-        if err != nil {
-            panic(err)
-        }
-        sonic_db_init = true
-    }
-}
-
-func init() {
-    sonic_db_init = false
+	if sonic_db_init {
+		return
+	}
+	ns_to_cfgfile_map := make(map[string]string)
+	// Ref: https://stackoverflow.com/questions/14928826/passing-pointers-to-maps-in-golang
+	DbGetNamespaceAndConfigFile(ns_to_cfgfile_map)
```

Comment:
Seems that logic inside this function can be executed during the auto package initialization with `init`, and it can simplify code and remove 
```
if !sonic_db_init {
    DbInit()	
}
```
from each function call. 
Do we really need to skip possible errors with config files and handle them at runtime?
Or probably it's better to explicitly show errors and panic on library loading and avoid run lib with invalid config.

Reply:
existing design. not changing as part of this PR.